javascript/extractor/lib/typescript/src/main.ts

```diff
@@ -529,26 +532,37 @@ function getEnvironmentVariable<T>(name: string, parse: (x: string) => T, defaul
     return value != null ? parse(value) : defaultValue;
 }
 
+/**
+ * Whether the memory usage was last observed to be above the threshold for restarting the TypeScript compiler.
+ */
+let isAboveReloadThreshold = false;
```

Comment:
I'm not fond of this naming/doc.  
The variable protects against reloading the project multiple times, and I don't think that is immediately clear from the naming. 

How about renaming to something like `hasReloadedSinceExceedingThreshold` (or the shorter `hasReloadedProject`)?